### PR TITLE
Move port utility out of endpoints controller

### PR DESCRIPTION
**Pull request by 0xmichalis**
The logic in here is pretty re-usable that it seems natural to move it into a separate util package. 

cc: @pweil-

### CODE REVIEW

File: pkg/util/port/port_find.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+// FindPort locates the container port for the given pod and portName.  If the
+// targetPort is a number, use that.  If the targetPort is a string, look that
+// string up in all named ports in all containers in the target pod.  If no
+// match is found, fail.
+func FindPort(pod *api.Pod, svcPort *api.ServicePort) (int, error) {
```
Comment (thockin) [inline pkg/util/port/port_find.go]: IF we do this, generify so it takes an IntOrString (portName) rather than the svcPort.

**Comment by thockin** [comment]
Do you have a use for this?  I'm tryin to resist the urge to make everything halfway-reusable into a util.  I prefer that util really be things that have no kubernetes tie-ins (we're not there yet but getting better).

If we actually have a use for this, I might see it in `pkg/api/util/pod/` - we have something similar for endpoints logic that is used in a few places.

**Comment by kargakis** [comment]
> Do you have a use for this?

Yes, we do: https://github.com/openshift/origin/pull/5578. We have to infer the service target port number from the container ports (in case of services with a named target port) which port in turn we set to an Openshift route that is going to be generated by `oc expose` and this utility is exactly what we need.

File: pkg/util/port/port_find.go (hunk 1 above)
Comment (kargakis) [inline pkg/util/port/port_find.go]: The protocol of the service port is used in here too.

**Comment by kargakis** [comment]
Moved under pkg/api/util/pod

**Comment by thockin** [comment]
I told you the wrong thing.  I meant `pkg/api/pod` - to echo `pkg/api/endpoints`

Otherwise LGTM

**Comment by kargakis** [comment]
@thockin moved in pkg/api/endpoints/util.go

**Comment by thockin** [comment]
So close.  You want `pkg/api/pod/util.go` - `endpoints` is for Endpoints utils.

**Comment by kargakis** [comment]
ok

**Comment by kargakis** [comment]
@thockin anything else here?

**Comment by kargakis** [comment]
> @thockin anything else here?

On your way out please add a cla human approved tag

File: pkg/api/pod/util.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+// FindPort locates the container port for the given pod and portName.  If the
+// targetPort is a number, use that.  If the targetPort is a string, look that
+// string up in all named ports in all containers in the target pod.  If no
+// match is found, fail.
+func FindPort(pod *api.Pod, svcPort *api.ServicePort) (int, error) {
```
Comment (thockin) [inline pkg/api/pod/util.go]: rather than receiving a ServicePort, this should just receive an IntOrString.  We can do this in a followup :)

Comment (thockin) [inline pkg/api/pod/util.go]: nevermind, I see why it can't

Comment (thockin) [inline pkg/api/pod/util.go]: actually - it COULD take an IntOrString and a protocol.

Comment (kargakis) [inline pkg/api/pod/util.go]: follow-up

**Comment by kargakis** [comment]
e2e tests on jenkins need to be rekicked

**Comment by gmarek** [comment]
@k8s-bot test this

**Comment by thockin** [comment]
@k8s-bot test this

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by kargakis** [comment]
2403 tests run, 8 skipped, 0 failed. 

And it's still red -__-

**Comment by thockin** [comment]
@k8s-bot unit test this please

**Comment by thockin** [comment]
@k8s-bot unit test this please

@kargakis did you run unit and integration tests yourself, just to be sure?

**Comment by thockin** [comment]
needs a rebase, sorry

**Comment by kargakis** [comment]
Rebased

> @kargakis did you run unit and integration tests yourself, just to be sure?

Both are clean for me
http://pastebin.com/NNz6cRax
http://pastebin.com/AaegE4v2

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Pull request merged**